scripts/analyze

```diff
@@ -0,0 +1,168 @@
+
+force = ['']
+featurefile = ''
+
+for i in sys.argv:
+    if i == sys.argv[0]:
+        continue
+    option = i.split('=')[0]
+    value = i.split('=')[1]
+    if option == 'BOARD':
+        board = value
+    elif option == 'SCRIPT':
+        script = value
+    elif option == 'FORCE':
+        force = value.split(',')
+    elif option == 'PRJBASE':
+        prjbase = value
+    elif option == 'PRJCONF':
+        prjconf = value
+    elif option == 'MAKEFILE':
+        makefile = value
+    elif option == 'MAKEBASE':
+        makebase = value
+    elif option == 'PROFILE':
+        featurefile = value
+
+print("Analyzing %s for %s" % (script, board))
+
+dependencies = []
+zephyr_config = []
+src_list = []
+zjs_config = []
+jrs_config = []
+
+def write_jrsconfig(list):
+    if featurefile != '':
+        if os.path.exists(featurefile):
+            shutil.copyfile(featurefile, featurefile + '.bak')
+        with open("fragments/jerry_feature.base", "r") as jrs:
+            feature = jrs.read()
+        with open(featurefile, "w") as p:
+            for i in list:
+                feature = feature.replace(i, "#" + i)
+            p.write(feature)
+
+def write_makefile(list, file):
+    with open(makebase, "r") as b:
+        base = b.read()
+    with open(file, "w") as f:
+        f.write(base)
+        if list:
+            f.write("obj-y += \\\n")
+            for i in list:
+                f.write("\t" + i.replace(".c", ".o") + " \\\n")
+            f.write("\nccflags-y += ")
+            for i in zjs_config:
+                f.write(i + " ")
+            f.write("\n")
+
+def write_zconf(list, file):
+    with open(file, "w") as f:
+        for i in list:
+            f.write(i + '\n')
+
+def add_to_list(list1, list2):
+    if list1 == None:
+        return list2
+    if list2 == None:
+        return list1
+    list1_set = set(list1)
+    list2_set = set(list2)
+    diff = list2_set - list1_set
+    return list1 + list(diff)
+
+def find_deps(name):
+    global dependencies
+    global zephyr_config
+    global src_list
+    global zjs_config
+    global jrs_config
+    with open("src/zjs_" + name + ".json") as f:
+        data = json.load(f)
+    if 'depends' in data:
+        depends = data['depends']
+        dependencies = add_to_list(dependencies, depends)
+        for i in depends:
+            l = find_deps(i)
+            dependencies = add_to_list(l, dependencies)
+    if 'targets' in data:
+        found = False
+        for i in data['targets']:
+            if i == board:
+                found = True
+                break
+        if not found:
+            return
+    if 'zephyr_conf' in data:
+        z_conf = data['zephyr_conf']
+        if 'all' in z_conf:
+            zephyr_config = add_to_list(zephyr_config, z_conf['all'])
+        if board in z_conf:
+            zephyr_config = add_to_list(zephyr_config, z_conf[board])
+    if 'zjs_config' in data:
+        zjs_config = add_to_list(zjs_config, data['zjs_config'])
```

Comment:
You should probably make this edit the first list in place, rather than return it. Lists are mutable in Python. Really you haven't defined "add_to_list" (append) but "merge_lists".

If you do that, these two lines could become:
```python
add_to_list(zjs_config, data.get('zjs_config', []))
```

Same can be all over the place to trim things down.